### ISSUE TITLE
State  лучше оставлять в корне вьюхи

### ISSUE THREAD

**Issue by ermolnik** [AlekseyYakovlev/Coroutines#1]
https://github.com/AlekseyYakovlev/Coroutines/blob/ddc06b8e5ec409751dd0881c55cb3e43e539b96d/news/src/main/java/ru/ermolnik/news/NewsScreen.kt#L37

Получилось довольно интересно, рулить стейтом внутри списка. 
В большинстве случаев это будет тяжелее поддерживать, нежели у тебя when(state) лежал в корне экрана

Например, если тебе нужно будет добавить диалог/BottomSheet/какой-то верхнеуровневый элемент тебе придется дважды делать when(state)
Ну и банально разбираться во вложенном коде сложнее, чем с ситуацией когда у тебя просто перебор стейтов и к ним вьюшки

**Comment by AlekseyYakovlev** [comment]
Согласен, но сильно зависит от состава стейта, написан ли он в виде дата класса с набором полей или силд класса описывающим конкретные состояния.
В данном конкретном случае оптимальным расположением when(state) считаю внутри SwipeRefresh, т.к. общий стейт используется в данном виджете для отображения индикатора загрузки.
Коммит включающий данные изменения: [https://github.com/AlekseyYakovlev/Coroutines/commit/a0c240205eda404899944e65aacbcd48fff72a77](url)
